opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -114,6 +114,25 @@ def shutdown(self) -> None:
         for sp in self._span_processors:
             sp.shutdown()
 
+    def force_flush(self, timeout_millis: int = None) -> bool:
+        flushed_within_time = True
+        flush_start_ns = time_ns()
+        for sp in self._span_processors:
+            if timeout_millis is None:
+                flushed_within_time = sp.force_flush() and flushed_within_time
+                continue
+
+            remaining_time_millis = (
```

Comment:
I think it would be better to calculate `deadline = time_ns + timeout_millis * 1000000` first and then compare against that instead of repeatedly subtracting.

Reply:
without subtracting it might lead to an overall timeout of about 2 times the given `timeout_millis`.
E.g. having 2 span processors in the `MultiSpanProcessor` and the 1st one flushing for almost `timeout_millis`. so the deadline wouldn't have been reached and the 2nd span processor would start flushing with a timout of `timeout_millis`.

Reply:
For the timeout provided to the force_flush calls made here, you can just use `deadline - time_ns()`. I guess you could also call that "repeatedly subtracting", but that way it does not accumulate errors from reusing the results of previous subtractions.